From 1f49be4429c17475b789e9089ce4d0ae48315e74 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Thu, 16 Feb 2017 09:15:29 +0100
Subject: [PATCH 76/79] FILES: Remove unnecessary check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

"grp_iter->gr_mem" is an array of strings and not just a string.
We tried to compare first string to NULL (acctually '\0')
But after that we iterated over the array to find count of members
and we check for NULL one more time.

Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
---
 src/providers/files/files_ops.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/providers/files/files_ops.c b/src/providers/files/files_ops.c
index 9ebf3b11bdb6be60c03392a74a0d434c90808782..370dcd971459069915178bac1fa3e87ecf2884ef 100644
--- a/src/providers/files/files_ops.c
+++ b/src/providers/files/files_ops.c
@@ -189,7 +189,7 @@ static errno_t enum_files_groups(TALLOC_CTX *mem_ctx,
         }
         grp->gr_passwd = talloc_strdup(grp, grp_iter->gr_passwd);
 
-        if (grp_iter->gr_mem != NULL && grp_iter->gr_mem[0] != '\0') {
+        if (grp_iter->gr_mem != NULL) {
             size_t nmem;
 
             for (nmem = 0; grp_iter->gr_mem[nmem] != NULL; nmem++);
-- 
2.9.3

